### PR TITLE
yarn2nix build

### PR DESCRIPTION
**Pull request by globin**
This is the missing piece to fully create a linux installer after @cleverca22's work.
`nix-shell` support will come from the `shell.nix` on @cleverca22's branch.

Some notes:
 * nixpkgs bump to include nodejs headers (https://github.com/NixOS/nixpkgs/commit/fda4b93cd4fd3775408117c380aab0f33737d30f)
 * removes web3 as it isn't used AFAICS and includes an (in yarn2nix) not supported git dependency

### CODE REVIEW

File: nixpkgs-src.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "owner":  "NixOS",
     "repo":   "nixpkgs",
-    "rev":    "ffea68a09cedab941d19f02c9041689ebc81192e",
-    "sha256": "1ipcgq111d11q78z9isfc00ndh7925asjkdjmb41ngfi9pabi0x3"
+    "rev":    "fda4b93cd4fd3775408117c380aab0f33737d30f",
```
Comment (rvl) [inline nixpkgs-src.json]: I'm fine with 18.03 if everyone else is fine with 18.03. Seems like this is a good enough reason to change branch.

File: default.nix [hunk 2]
```diff
@@ -1,16 +1,32 @@
...
-    gnumake
-  ];
+with (import (fetchTarball https://github.com/moretea/yarn2nix/archive/v1.0.0.tar.gz) { inherit pkgs nodejs; });
 
-  src = null;
-
-}
+pkgs.callPackage (
+{ stdenv, python }:
+mkYarnPackage {
```
Comment (rvl) [inline default.nix]: Is "impure development shell" a requirement? e.g. a nix-shell where developers can run yarn, npm, etc.
If so does `mkYarnPackage` support that?

File: default.nix [hunk 3]
```diff
@@ -1,16 +1,32 @@
...
-stdenv.mkDerivation {
-  name = "daedalus";
-
-  buildInputs = [
-    nix bash binutils coreutils curl
-    git python27 curl electron nodejs-8_x
-    nodePackages.node-gyp nodePackages.node-pre-gyp
-    gnumake
-  ];
+with (import (fetchTarball https://github.com/moretea/yarn2nix/archive/v1.0.0.tar.gz) { inherit pkgs nodejs; });
```
Comment (rvl) [inline default.nix]: Will this be in nixpkgs eventually?

File: default.nix (hunk 2 above)
Comment (globin) [reply default.nix]: @cleverca22 is planning on merging with his branch, that includes a `shell.nix` for this.

File: default.nix (hunk 3 above)
Comment (globin) [reply default.nix]: It is on nixpkgs master (https://github.com/NixOS/nixpkgs/pull/35340)

**Comment by globin** [comment]
Only build-time dependencies that run native code need special handling, that should hardly ever occur (node-sass obviously is an exception to that).

**Comment by domenkozar** [comment]
@cleverca22 this is now merged?

**Comment by cleverca22** [comment]
@domenkozar it has been merged into the development-webpack-fork branch

**Comment by domenkozar** [comment]
Thanks!

**Pull request closed without merging**